#! /bin/sh
patch -p0 -f $* < $0
exit $?

2010-09-24  Werner Koch  <wk@g10code.com>

	* gpg-agent.c (main, reread_configuration): Always test whether
	the default configuration file has been created in the meantime.
	Fixes bug#1285.


--- agent/gpg-agent.c	(revision 5367)
+++ agent/gpg-agent.c	(working copy)
@@ -724,6 +724,12 @@
               if( parse_debug )
                 log_info (_("NOTE: no default option file `%s'\n"),
                           configname );
+              /* Save the default conf file name so that
+                 reread_configuration is able to test whether the
+                 config file has been created in the meantime.  */
+              xfree (config_filename);
+              config_filename = configname;
+              configname = NULL;
 	    }
           else
             {
@@ -811,10 +817,15 @@
       fclose( configfp );
       configfp = NULL;
       /* Keep a copy of the name so that it can be read on SIGHUP. */
-      config_filename = configname;
+      if (config_filename != configname)
+        {
+          xfree (config_filename);
+          config_filename = configname;
+        }
       configname = NULL;
       goto next_pass;
     }
+    
   xfree (configname);
   configname = NULL;
   if (log_get_errorcount(0))
@@ -1332,8 +1343,8 @@
   fp = fopen (config_filename, "r");
   if (!fp)
     {
-      log_error (_("option file `%s': %s\n"),
-                 config_filename, strerror(errno) );
+      log_info (_("option file `%s': %s\n"),
+                config_filename, strerror(errno) );
       return;
     }
 

